src/main/java/dinkplugin/notifiers/DeathNotifier.java

```diff
@@ -33,57 +42,83 @@ public void onActorDeath(ActorDeath actor) {
+            lastTarget = new WeakReference<>(event.getTarget());
         }
-        ItemContainer inventory = plugin.getClient().getItemContainer(InventoryID.INVENTORY);
-        ItemContainer equipment = plugin.getClient().getItemContainer(InventoryID.EQUIPMENT);
+    }
 
-        assert inventory != null;
-        assert equipment != null;
-        List<Pair<Item, Long>> itemsByPrice = Stream.concat(Arrays.stream(inventory.getItems()), Arrays.stream(equipment.getItems()))
-            .map(item -> Pair.of(item, (long) (plugin.getItemManager().getItemPrice(item.getId())) * (long) (item.getQuantity())))
-            .sorted((item1, item2) -> Math.toIntExact(item2.getRight() - item1.getRight()))
-            .collect(Collectors.toList());
+    private void handleNotify() {
+        Player localPlayer = plugin.getClient().getLocalPlayer();
+        Actor pker = identifyPker();
 
-        int keepCount = 3;
-        if (localPlayer.getSkullIcon() != null) {
-            keepCount = 0;
-        }
-        if (plugin.getClient().isPrayerActive(Prayer.PROTECT_ITEM)) {
-            keepCount += 1;
-        }
-        Long losePrice = itemsByPrice.stream()
+        List<Pair<Item, Long>> itemsByPrice = getPricedItems();
+        int keepCount = (localPlayer.getSkullIcon() == null ? 3 : 0) +
+            (plugin.getClient().isPrayerActive(Prayer.PROTECT_ITEM) ? 1 : 0);
+        long losePrice = itemsByPrice.stream()
             .skip(keepCount)
             .map(Pair::getRight)
             .reduce(Long::sum)
             .orElse(0L);
 
-        String template = plugin.getConfig().deathNotifyMessage();
-        if (pker != null && plugin.getConfig().deathNotifPvpEnabled()) {
-            template = plugin.getConfig().deathNotifPvpMessage();
-        }
+        String template = pker != null && plugin.getConfig().deathNotifPvpEnabled()
+            ? plugin.getConfig().deathNotifPvpMessage()
+            : plugin.getConfig().deathNotifyMessage();
         String notifyMessage = template
             .replace("%USERNAME%", Utils.getPlayerName(plugin.getClient()))
-            .replace("%VALUELOST%", losePrice.toString());
+            .replace("%VALUELOST%", String.valueOf(losePrice));
         if (pker != null && plugin.getConfig().deathNotifPvpEnabled()) {
-            // player name hopefully isn't null
-            notifyMessage = notifyMessage
-                .replace("%PKER%", Objects.requireNonNull(pker.getName()));
+            notifyMessage = notifyMessage.replace("%PKER%", String.valueOf(pker.getName()));
         }
 
+        List<NotificationBody.Embed> lostItemEmbeds = itemsByPrice.stream()
+            .skip(keepCount)
+            .limit(3)
+            .map(Pair::getLeft)
+            .mapToInt(Item::getId)
+            .mapToObj(Utils::getItemImageUrl)
+            .map(NotificationBody.UrlEmbed::new)
+            .map(NotificationBody.Embed::new)
+            .collect(Collectors.toList());
```

Comment:
Since the feature is unreleased this is ok to be under the same changelog entry

Reply:
if desired, can be added as a minor entry (since it also appears in non-pvp situations)

Reply:
Not needed imo, this should be all good :+1: